feat: support setting secrets for renovate

gmariotti opened the pull request:
Hi, I created this PR to make it possible to use k8s/Secret to pass sensitive data to Renovate. I'm no Helm expert, I probably missed something for sure 😅

gmariotti commented:
The tests expect that I bump the version of the chart but the version seems to be bounded to the version of renovate. What should I do?

JamieMagee commented:
@gmariotti Sorry I've missed this PR. I'll find some time today to review it.

We decided to pin the helm chart version to the renovate version to make auto updates easier. I think we should be able to add a build version, and still stay compliant with semver. Something like `19.239.11+1`

JamieMagee commented on charts/renovate/templates/secret.yaml:
```diff
@@ -0,0 +1,16 @@
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+type: Opaque
+data:
```
I would simplify this to `stringData` and drop the `b64enc`

gmariotti replied:
I will fix it, thanks 🙏

gmariotti replied:
I didn't find any `stringData` function, could you point me to it?

gmariotti commented:
What about documentation? Is there anything I need to do?

JamieMagee commented:
Yeah, can you add an example to the `README.md` in the root of the repo, as well as the `README.md.gotmpl` and `README.md` in the `charts/renovate` directory.

And don't worry about merging, I'll figure it out 😅

gmariotti commented:
@viceice @JamieMagee rebased with master and addressed some of the comments, but I'm still having a problem with the version number (probably because of the commit prefix used), any suggestion?

viceice commented on charts/renovate/Chart.yaml:
```diff
@@ -1,5 +1,5 @@
+appVersion: '22.25.6+1'
 description: Universal dependency update tool that fits into your workflows.
 name: renovate
 version: '22.25.6'
```
```suggestion
version: '22.25.6+1'
```

viceice commented on charts/renovate/Chart.yaml:
```diff
@@ -1,5 +1,5 @@
 apiVersion: v1
-appVersion: '22.25.6'
+appVersion: '22.25.6+1'
```
```suggestion
appVersion: '22.25.6'
```

viceice commented:
Should be released with next renovate update

viceice commented:
v23 on the way: renovatebot/docker-renovate-full#59
